TMEDIA-788 : Style the components for Commerce theme & TMEDIA-767 : Alias Tokens

ecarlisle opened the pull request:
## Tickets

- [TMEDIA-788](https://arcpublishing.atlassian.net/browse/TMEDIA-788)
- [TMEDIA-767](https://arcpublishing.atlassian.net/browse/TMEDIA-767)

## Description

Styling for components has been updated as per design specification in:
1.  [Figma](https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?node-id=2721%3A19065)
2. [Confluence](https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3118465081/Design+Tokens)

PR in `arc-themes-feature-pack` will be updated and merged with these styles upon approval in this repository:
https://github.com/WPMedia/arc-themes-feature-pack/pull/224

## Acceptance Criteria

**From TMEDIA-788:**
1. Use the latest agreed-upon global token and alias definitions
3. Update the default styles for each component
4. Update Storybook

**From TMEDIA-767:**
An alias Scss map that contains:
1. Alias key names (Friendly naming) - See Figma resource for alias tokens currently used by design
2. Names should be split by a hyphen
3. Initial values that reference global tokens where possible
4. Font Family is likely to be an exception to the rule here

## Known Inconsistencies
1. The pill component has been given a `border-radius` of `2em`. Currently there is not a `border-radius` token of this size.
2. In Figma the Pill component has a top/bottom padding of `6px` which is not a token value. I rounded up to `8px` which is `global-spacing-2`.
<img width="178" alt="image" src="https://user-images.githubusercontent.com/452134/166569582-0191d9c3-a014-4b27-a713-e047851888fc.png">
^^ Figma Spec

## Test Steps
1. Checkout branch - `git checkout TMEDIA-788-updated`
5. Update dependencies - `npm i`
6. Run Storybook `npm run storybook`
7. Review component styling in Storybook UI.
8. **Note:** `global-spacing-token-2` has been added with a value of `8px`. All other existing spacing tokens over `global-spacing-token-2` have been moved up to the next token scale (we now have 17 spacing tokens instead of 16). All tokens using spacing tokens over 2 have been reassigned to account for this change.

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed relevant documentation has been updated/added.
- [ ] Add label - **ready for review** when the pull request is ready for someone to begin reviewing

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] All GitHub Actions pass
- [ ] Ran the code locally based on the test instructions.
- [ ] Checked Chromatic for Storybook changes, accepted the updates if acceptable
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.
- [ ] Approve and Add label - **ready to merge** if you are happy with the pull request
- [ ] Want another reviewer? Add the label **additional review**

ecarlisle committed:
Insert Google Fonts for Storybook <head>.

ecarlisle committed:
Update styles for Commerce components.

RobinGiannattasio commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+		// alignment.  It accounts for a minimum width that has (--global-spacing-5 / 2) margins
+		//
+		"layout-content-scale-width":
+			min(
+				calc(100vw - var(--global-spacing-5)),
+				calc(var(--layout-content-max-width) / var(--layout-max-width) * 100%)
+			),
```
These types of values are going to be very interesting to consider in how we approach them in Theme Settings 🤔

ecarlisle replied:
I share this concern. This came in with the tokens file from the `arc-themes-feature-pack` repository.
https://github.com/WPMedia/arc-themes-feature-pack/blob/commerce/site-styles/arc-commerce/_index.scss
It was that file to maintain source of truth and prevent fragmentation in token usage.

RobinGiannattasio replied:
Cool cool. I'll bring it up with the themes core team so we can make sure we have this example when we do our solution designs.

RobinGiannattasio replied:
Actually - lemme just tag 'em.

@jgrosspietsch @annasherman ^ Peep this 👀

brentmiller1973 replied:
@ecarlisle Wondering if this might be a good use case for clamp?  Would allow client to not only have the min functionality, but the max as well in one definition

ecarlisle replied:
In addition to handling a token value with a calculation, there are two other cases I wanted to draw attention to as think about handling in Themes Settings:
1. Unitless values (see lines 157, 158 above)
2. CSS variables used for component functionality (see `--slides:` for carousel).
While these may be needed for component and/or block design, we'll probably want to consider the UX for those  cases.

RobinGiannattasio commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+			),
+		"primary-color": var(--global-neutral-10),
+		"primary-font-family": "Inter",
+		"secondary-font-color": "Lora",
```
Assume you mean `secondary-font-family` here.

ecarlisle replied:
LOL, indeed! :-D
Corrected.

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+					),
+					button-secondary-reverse-hover: (
+						border-color: var(--global-neutral-7),
+						background-color: transparent,
```
No need for `background-color` here as it will inherit the same value from the non hover state

ecarlisle replied:
Thanks! I've removed the `background-color`.

JackHowa replied:
removed

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+							text-decoration: underline,
+						),
+					button-primary: (
+						background-color: var(--global-neutral-10),
```
We can and should be using the `primary-color` alias token here - Primary Button is aligned to the primary color for our designs

ecarlisle replied:
Thanks! I've removed the `primary-color`.

JackHowa replied:
ok. I've addressed this in pr. did you follow what Matthew was saying here and the change I made? it looks like you said you were going to remove the primary-color when he said to use it in more places

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+			components:
+				(
+					body: (
+						background-color: var(--global-color-white),
```
We don't need to explicitly set the background color if it is white as that is the browser default

ecarlisle replied:
Thanks! I've removed the `background-color` property.

JackHowa replied:
removed

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+							background-color: var(--global-white),
+						),
+					button-default-hover: (
+						color: var(--global-neutral-10),
```
No need to set `color` as it is the same value as the non hover state

ecarlisle replied:
Thanks! I've removed `color`.

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+					link-active: (
+						color: var(--global-neutral-10),
+					),
+					link-visited: (
```
Are we awaiting design decision on the link hover, active and visited colors? Currently we don't need to redefine the same color for all states as we can rely on CSS inheritance here

ecarlisle replied:
I removed them, but was wondering about browser defaults for `visited` and `active`, as sometimes this can be purple.

nschubach replied:
I think @ecarlisle brings up a valid point here.  Unless we want to handle that situation in the reset.  @JackHowa thoughts?

JackHowa replied:
if this is not in the designs, then we should hold off for the sake of this ticket. Move quick based on existing designs

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+					//
+					// ENGINEERING & DESIGN QUESTION:
+					// 1. Engineering has a "Pill" component.
+					// 2. This is represented in Figma as a "Badge."
```
We should be following the Badge figma component

<img width="779" alt="Figma-2022-04-27-10-14 18" src="https://user-images.githubusercontent.com/868127/165484941-46e6e406-c97b-46b6-97f6-5f107c0b4439.png">

JackHowa replied:
looks like this question can be answered that the badge figma component is the pill

JackHowa replied:
we're using the badge/pill figma component. formerly called a badge

![Screen Shot 2022-05-02 at 13 17 41](https://user-images.githubusercontent.com/5950956/166302652-ea1434da-edda-44c5-9809-f4c324a81c9c.png)

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+							// We don't have a border radius token yet that will cause
+							// The left and right borders to curve completely.
+							// Do we need one?
+							border-radius: 0.8rem,
```
Currently we only have 1 global border radius token - this is a good opportunity to expand the range of the border radius tokens

ecarlisle replied:
cc: @brannytempy

JackHowa replied:
@ecarlisle can you add tokens for this? cc @brannytempy @haganka

JackHowa replied:
@ecarlisle has this been addressed?

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+					),
+					// DESIGN QUESTION:
+					// There are multiple paddings for input in Figma.
+					// Is this the correct one for a standard, default text input?
```
I see below you are making use of `margin-top` and `margin-bottom` values. We should avoid using `margin`s to do spacing, and use `flex` with gap to ensure consistent spacing. We can then see the spacing on the input as a whole and any children would get the same spacing applied

```
input: (
    align-items: flex-start,
	display: flex,
	flex-direction: column,
	gap: var(--global-spacing-1),
),
```

ecarlisle replied:
Good call! I've added this input selector and removed the display and margin properties.

JackHowa replied:
has this been addressed @ecarlisle ?

JackHowa replied:
yes, this has ^^

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+						font-weight: var(--global-font-weight-7),
+						margin-bottom: var(--global-spacing-1),
+					),
+					input-tip: (
```
We could remove the need to set `input-tip` value defaults if we use flex and gap - see previous comment

ecarlisle replied:
Removed.

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+					input-label: (
+						display: block,
+						font-weight: var(--global-font-weight-7),
+						margin-bottom: var(--global-spacing-1),
```
We should use `gap` with flex see previous comment

ecarlisle replied:
Corrected.

JackHowa replied:
I don't see gap used on the inputs

JackHowa replied:
looks like this has been addressed

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+							outline: 3px double var(--global-neutral-5),
+						),
+					input-label: (
+						display: block,
```
No need for `display: block` if we use flex

ecarlisle replied:
Corrected.

JackHowa replied:
can you address this @ecarlisle ?

JackHowa replied:
addressed

matthewroach commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+						(
+							border: 1px solid var(--global-neutral-5),
+							border-radius: var(--global-border-radius-1),
+							display: block,
```
No need for `display: block` if we use flex

ecarlisle replied:
Corrected.

ecarlisle committed:
Correct secondary font property name.

ecarlisle committed:
Remove input-tip selector.

brentmiller1973 commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+					// DESIGN QUESTION:
+					// There are multiple paddings for input in Figma.
+					// Is this the correct one for a standard, default text input?
+					input-input:
```
`input-input` This is confusing.

ecarlisle commented:
@matthewroach : To this comment...
https://github.com/WPMedia/arc-themes-components/pull/84#discussion_r859568878
... I've updated the Pill component styles as per the examples you gave in the image. Thanks!

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,577 @@
-					border-bottom-style: solid,
-					border-bottom-color: #712d2d,
-					height: 1px,
+			// DESIGN ASSUMPTION: Attribution does not require component styling.
```
interesting looks like the comment is causing the new line due to a known prettier bug. no worries -- once the comments/questions are done, they should go away  

playground
https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABArgZzgAgDogB0hA9MSZmJpoZdQTgO4CWMAFjgBR5Q584kSOAGY8+eGABIAhgBtm0zEi48JakLOkBPCOhgBaALbSAHvpYATdgWUBGACz2ADABpVMdZp179kWAgNjM0trZBxbACZbADY3XnwPDw1tXUDTc2YrNhtwx1d3TxSfP3hYI3SQ7LDImLjxRIIvVPLgzNC7PLi1JKbi6FK01qycmtixHABKF3x4mSgIdjgAJ305BUwM9n15yENDAOVueO7GosGM4erOgp6z336Alov28Kix44bk73PKkeuoE5fZolR5BZ5VOxvOoJQrfJ6-K7OLqfXoGEFlMEIyG1cYTXEgFwgCAABxgzGgmGQoGkSyWEEYAAUaQhKShpAA3CCZAkgABGS2kYAA1nAYABlaT7AAyzCgcGQwjk2EJ-MFIvFxMFsoA5sgYEt0HBCXBDLy4BYLOapdIoNr0NJtXAAGIQJbGGBk23IEDSPQQHlsGCGWQAdTYrDgmE1YDgYpZrGY7NYWm9VEphNl2CWMAZAu1xgVSqNIAAVpgTGKdbI4ABFdALeVIRWyZUgTVLLPeujpttLWUwENtNjIAAcrl7EGwIYFxO9xKWkeW7PlhIAjvX4LmSayfRs5ebzTyF+vmAvcw6C02i4TsIZmHqDcXMFXaxvG83WzBpLzB1lkBFCX1aRmHkW0AGEID2aRvXmOUeSwOAABVv1ZD9i3ZQ0AEkoEtWAxTAPtSQAQRwsUYC0atCxbYsyxMAAhAVhVFCVpVld8iwAXw4oA

https://github.com/prettier/prettier/issues/4659

JackHowa replied:
if we remove the comment here, then the formatting will be as before `:(`

ecarlisle replied:
@JackHowa : Yes, I put in the comments to ask questions and will remove them when those are resolved. It's interesting to hear about this bug, though. I've also noticing that comments on the same line are being saved to the line below (see font-size, spacing, line-height). This is confusing for me & wondering if it's caused by the same bug.

ecarlisle replied:
BTW, sorry, didn't mean to resolve this discussion.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,577 @@
+				calc(100vw - var(--global-spacing-5)),
+				calc(var(--layout-content-max-width) / var(--layout-max-width) * 100%)
+			),
+		"primary-color": var(--global-neutral-10),
```
secondaryColor and other alias tokens are out of scope here, right?

ecarlisle replied:
Correct. Those will come out of TMEDIA-767 & TBRANDS-43.

JackHowa replied:
out of scope

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,577 @@
+					// color tokens. I took my best guess with global-red-4.
+					input-error-input:
+						(
+							border-color: var(--global-red-4),
```
looks like the border color is alias token form danger #B5646

Yeah I couldn't find that in the reds either. red-4 is porbably the closest with b5... 

![Screen Shot 2022-04-27 at 14 26 04](https://user-images.githubusercontent.com/5950956/165604468-2699bc72-3f42-4989-a2d3-c1ba4944cf86.png)

ecarlisle replied:
Yes. I believe @brannytempy and @haganka are working on this.
I put in notes for best estimates in the meantime.

ecarlisle replied:
Updated with global color values provided by Design.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,577 @@
+						height: var(--global-spacing-4),
+						width: var(--global-spacing-4),
+					),
+					input: (
```
is there a form warning focus state? Looks like that #c986... orange is another color that isn't on the orange scale but exists as an alias token

![Screen Shot 2022-04-27 at 14 28 50](https://user-images.githubusercontent.com/5950956/165604945-20db6b43-95d7-40e4-aa6a-5c04cf274eef.png)

![Screen Shot 2022-04-27 at 14 29 46](https://user-images.githubusercontent.com/5950956/165605193-fecc791d-c525-493c-ba9c-42d371c105a9.png)

ecarlisle replied:
@JackHowa : Thanks, good catch!
I added the tokens for warning and success with focus tokens. There aren't exact matches on the colors but I've put questions in the comments.

ecarlisle replied:
**Update:** I've updated all the input colors with global color values confirmed by design.

JackHowa replied:
cool and design approves, it seems

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,577 @@
 			),
 		),
 		desktop: (
+			alias: (),
```
this alias can be removed for desktop as it's empty. Probably a move for another day having breakpoint-dependent aliases. I suppose it's a style decision but it's not necessary when I removed it

ecarlisle replied:
@JackHowa : Good call! Removed.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,577 @@
 			),
 		),
 		desktop: (
+			alias: (),
```
```suggestion
```

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,577 @@
-				carousel-additional-controls: (
-					display: block,
+				carousel: (
+					--slides: 4,
```
nice

JackHowa reviewed (commented):
Sorry for the confusion around the spacing. It looks like it's a prettier bug. Once we delete the comments asking questions, the styles and basic spacing will be fixed. 

Wasn't sure if you were missing warning state style on the focus of the input.

Looked like there were more alias tokens in the designs; I wasn't sure if those new colors (in some instances) needed to be added, like that red outline you asked about. `Use the latest agreed-upon global token and alias definitions`

ecarlisle committed:
Add input warning and success styling.

ecarlisle committed:
Correct pill padding.

ecarlisle committed:
Remove unneeded alias token for desktop.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,595 @@
-					border-bottom-style: solid,
-					border-bottom-color: #712d2d,
-					height: 1px,
+			// DESIGN ASSUMPTION: Attribution does not require component styling.
```
is this an open question? if so, has design addressed it. If not, we can remove this comment, right?

ecarlisle replied:
I'll put this in Slack to make sure it's understood. The "DESIGN ASSUMPTIONS" and "DESIGN QUESTION" comments are questions for Design.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,595 @@
+						line-height: var(--global-line-height-4),
+					),
+					// DESIGN QUESTION:
+					// Please confirm this style in Storybook for a link button.
```
is this a comment from design or for devs?

ecarlisle replied:
This one more so needs design confirmation.

ecarlisle replied:
Comment removed.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,595 @@
+					// DESIGN ASSUMPTION
+					// "Best match" assumptions were made w/ color and background
+					// color for the "default" this variant, based on both design in
+					// Zeplin as well as design trends in Figma. Please make any corrections.
```
is this a todo for design?

ecarlisle replied:
Yes.

ecarlisle replied:
Comment has been removed.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,595 @@
+					button-default-hover: (
+						background-color: var(--global-neutral-1),
+					),
+					// DESIGN ASSUMPTION: Date does not require component styling.
+					// DESIGN ASSUMPTION: Grid does not require component styling.
```
this can be removed, right?

ecarlisle replied:
Yes. Removed.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,595 @@
+						font-weight: var(--global-font-weight-7),
+					),
+					// DESIGN QUESTION:
+					// The input error border color in Figma is not one of the red
```
is this a todo? if so, can you add the new red tokens and remove this

ecarlisle replied:
Updated. Comment removed.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,595 @@
+					// color tokens. I took my best guess with global-orange-3.
+					input-warning-input:
+						(
+							border-color: var(--global-orange-3),
```
thanks for adding warning

ecarlisle replied:
Color token has been confirmed by design.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,595 @@
+							padding: var(--global-spacing-2) var(--global-spacing-3),
+						),
+					// DESIGN QUESTION:
+					// Do we have or need standard styling for the Price component
+					// with List Price and Sale Price?
+					//
+					// DESIGN ASSUMPTION: Separator does not require component styling.
+					//
+					// DESIGN ASSUMPTION: Stack does not require component styling.
+					//
+					// DESIGN ASSUMPTION: Video does not require component styling.
```
these can also be removed, right?

ecarlisle replied:
Yes. Removed.

JackHowa reviewed (changes requested):
Good work adding the form inputs missing. In general, I'm all for adding comments that improve my understanding of the work. But it seems like many of these comments are questions "What is the naming standard?" or notes to self "Attribution doesn't have any specific styles". These didn't help my understanding of the styles; if the team disagrees, then I'm ok with keeping them as they don't hurt anything. But it seems unnecessary to me unless this is a WIP. 

I see a lot of open questions for design still, including the alias tokens. I'm ok with adding a best guess color is that all the choice we have. But it looks like those tokens are available to add. Can you add them then? 

I also noticed some existing comments from Matthew that have not been addressed.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,595 @@
+					//
+					// DESIGN ASSUMPTION: Media Item does not require component styling.
+					//
+					// ENGINEERING & DESIGN QUESTION:
+					// 1. Engineering has a "Pill" component.
+					// 2. This is represented in Figma as a "Badge."
```
```suggestion
```

ecarlisle replied:
Updated. Comment is removed.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,595 @@
+					// ENGINEERING & DESIGN QUESTION:
+					// 1. Engineering has a "Pill" component.
+					// 2. This is represented in Figma as a "Badge."
+					// What naming convention are we using?
```
```suggestion
```

ecarlisle replied:
Updated. Comment is removed.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,595 @@
+							line-height: var(--global-spacing-2),
+							padding: var(--global-spacing-2) var(--global-spacing-3),
+						),
+					// DESIGN QUESTION:
+					// Do we have or need standard styling for the Price component
+					// with List Price and Sale Price?
+					//
+					// DESIGN ASSUMPTION: Separator does not require component styling.
+					//
+					// DESIGN ASSUMPTION: Stack does not require component styling.
+					//
+					// DESIGN ASSUMPTION: Video does not require component styling.
```
```suggestion
```

ecarlisle replied:
Updated. Removed.

ecarlisle committed:
Update input label not to have display property.

ecarlisle committed:
Update form success, warning, danger colors.

JackHowa commented:
things to do in pr 

- [ ] this 
- [x] that

JackHowa commented:
things to do 

- [ ] n
- [ ] y

ecarlisle committed:
Remove comment for Grid component.

ecarlisle committed:
Remove comment requesting color clarification.

ecarlisle committed:
Remove comment requesting clarification on Badge, Pill.

ecarlisle committed:
Remove comment requesting clarification on Video.

ecarlisle committed:
Remove comment requesting clarification on Price, Separator, and Stack.

ecarlisle committed:
Add scale # to all spacing tokens over 2 by 1.

ecarlisle committed:
Seeminly large change due to indentation change by Prettier.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,589 @@
+					button-link:
+						(
+							background-color: transparent,
+							color: var(--global-neutral-10),
```
```suggestion
							color: var(--primary-color),
```

looks like we should be using the alias token here, right? https://github.com/WPMedia/arc-themes-components/pull/84#discussion_r859563348

JackHowa replied:
made a pr making this fix throughout the file

JackHowa reviewed (changes requested):
Thanks for addressing design questions. Also it sounds like the color matches are right per design for the form inputs, for instance. I noticed that the comment to reuse primary-color had not been applied so I made a pr to do so https://github.com/WPMedia/arc-themes-components/pull/87

It also seemed like there were still some open questions about border-radius? https://github.com/WPMedia/arc-themes-components/pull/84/files/96e33321f5ec7e395a73d129f2026e7803fdc791#r859569724 

`The pill component has been given a border-radius of 2em. Currently there is not a border-radius token of this size.` -> can you add them?

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,543 @@
+		"z-index-3": 50,
+		"z-index-4": 100,
+		"z-index-5": 500,
+		"border-radius-1": 0.25em,
```
I bet we could iterate up more border radius aliases from here to start 👍

ecarlisle replied:
Agreed. A `border-radius` scale is an item pending from Design. I defer to them on determining the scale and number of tokens. If not by in TMEDIA-788, this will be covered in TMEDIA-767.

JackHowa replied:
potential follow-up ticket if this is not in scope of the 788 ticket -- but is 767. Create a ticket to follow-up with design pending 767.

ecarlisle replied:
I've added this as a sub-task of 767.

Commit:
refactor: Reuse alias token based on feedback (#87)

Per https://github.com/WPMedia/arc-themes-components/pull/84#discussion_r859563348

JackHowa commented:
- primary-color should be neutral 8 instead of neutral 10

brentmiller1973 commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,543 @@
+				button: (
+					border: 1px solid transparent,
+					border-radius: var(--global-border-radius-1),
+					font-size: var(--global-font-size-3),
```
Here is an example where Alias tokens can be expanded to create a bridge between global tokens and component and blocks.  Instead of setting font-size to `--global-font-size-3`, it is set to an alias token like `font-size: var(--button-font-size-3)`.  In return the alias token `--button-font-size-3` is assigned the global token `--global-font-size-3`

brentmiller1973 reviewed (approved):
Definitely want to have a deeper discussion on the use of alias tokens as a bridge between global styles and component/block styles.  See my comment in this pr at the button level.  This emphasis on alias tokens will make client onboarding easier, make themesettings engineering easier and also help to reduce component and block style over-riding.

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,543 @@
+					font-size: var(--global-font-size-2),
+					line-height: var(--global-line-height-2),
+				),
+				pill: (
```
In the designs, it looks like there's a success a few other states. This looks new from the designs but we don't have a success pill either 

https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?node-id=3505%3A27776

matthewroach replied:
The design's have been labelled incorrectly here - there is a need for a Pill and a Badge component - at the moment we do not have a "Badge" component that is currently called a Pill in the design document - our "Pill" component is what is currently being called a tag component

JackHowa commented on .storybook/themes/commerce.scss:
```diff
@@ -3,43 +3,543 @@
+				),
+				pill: (
+					background-color: var(--primary-color),
+					border-radius: 2em,
```
this font-weight should be bold 70
![Screen Shot 2022-05-02 at 12 53 15](https://user-images.githubusercontent.com/5950956/166299108-c539520a-956b-4ccf-92df-855b22491209.png)
0

JackHowa replied:
I've updated this @ecarlisle

matthewroach replied:
We should hold on this, until we get full clarification and alignment on the designs - as the component in this repository is aligned to the Tag component, we currently do not have a badge component

ecarlisle replied:
@matthewroach : The styles match the Badge component in Figma if that's the match we're looking for. Thanks!

matthewroach replied:
Figma has been updated and the Pill is now correctly labelled to the one I original linked you too

matthewroach replied:
`font-weight` should be `400`

JackHowa committed:
fix: Add font-weight for pill to be bold

JackHowa committed:
fix: Uncomment font-weight to be back to bold

JackHowa committed:
fix: Update the padding for horizontal pill

Should be 8 px. Was 16px

JackHowa commented:
Updating the padding based on pill mocks 
![Screen Shot 2022-05-02 at 13 07 58](https://user-images.githubusercontent.com/5950956/166301386-d4b6ee23-4a83-4050-9af0-7a1b0dd8d0fb.png)

JackHowa committed:
fix: Add text-color alias per mocks

ecarlisle committed:
Replace weight 400 font with 500 as per alias token specs.

ecarlisle committed:
Update w/ alias tokens and some mappings.

ecarlisle committed:
Update pill styles.

ecarlisle commented:
PR is outdated and in different code workflow path. Closing.
